src/recommendation/src/main/scala/RankingEvaluator.scala

```diff
@@ -0,0 +1,162 @@
+// Copyright (C) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in project root for information.
+
+package com.microsoft.ml.spark
+
+import org.apache.spark.ml.evaluation.Evaluator
+import org.apache.spark.ml.param._
+import org.apache.spark.ml.recommendation.{HasRecommenderCols, RecEvaluatorParams}
+import org.apache.spark.ml.util.{ComplexParamsReadable, ComplexParamsWritable, Identifiable}
+import org.apache.spark.mllib.evaluation.RankingMetrics
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.{Dataset, Row}
+
+import scala.reflect.ClassTag
+
+class AdvancedRankingMetrics[T: ClassTag](predictionAndLabels: RDD[(Array[T], Array[T])],
```

Comment:
is it necessary for this to be an RDD instead of a DataFrame?

Reply:
RankingMetrics requires RDD so I am not sure if this can be switched without creating a RankingMetrics DF version.

Reply:
Yes its necessary :/